src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp

```diff
@@ -0,0 +1,281 @@
+
+      // needed later for writing in ms file
+      int int_charge(1);
+
+      if (p == IonSource::Polarity::POSITIVE)
+      {
+        int_charge = +1;
+      }
+      else
+      {
+        LOG_WARN << "SiriusMSConverter (due to Sirius) only support positive ion mode and mono charged analytes." << endl;
+        continue;
+      }
+
+      //there should be only one precursor and MS2 should contain peaks to be considered
+      if (precursor.size() == 1 && spectrum.empty() == false)
+      {
+        //read charge annotated to MS2
+        int precursor_charge = precursor[0].getCharge();
+
+        //sirius only supports +1 charge so far
+        if (precursor_charge > 1 || precursor_charge <= -1)
+        {
+          LOG_WARN << "SiriusMSConverter (due to Sirius) only support positively mono charged analytes." << endl;
+        }
+
+        //get m/z and intensity of precursor != MS1 spectrum
+        double precursor_mz = precursor[0].getMZ();
+        float precursor_int = precursor[0].getIntensity();
+
+        //find corresponding ms1 spectra (precursor)
+        PeakMap::ConstIterator s_it2 = spectra.getPrecursorSpectrum(s_it);
+
+        double test_mz = precursor_mz;
+
+        vector<Peak1D> isotopes;
+        isotopes.clear();
+
+        if (s_it2->getMSLevel() != 1)
+        {
+          LOG_WARN << "Error: no MS1 spectrum for this precursor. No isotopes considered in sirius." << endl;
+        }
+          //get the precursor in the ms1 spectrum (highest intensity in the range of the precursor mz +- 0.1 Da
+        else
+        {
+          const MSSpectrum<Peak1D>& spectrum1 = *s_it2;
+
+          Int mono_index = getHighestIntensityPeakInMZRange(test_mz, spectrum1, 0.2, 0.2);
+
+          if (mono_index != -1)
+          {
+            const Peak1D& max_mono_peak = spectrum1[mono_index];
+            isotopes.push_back(max_mono_peak);
+
+            // make sure the 13C isotopic peak is picked up by doubling the (fractional) mass difference (approx. 1.0066)
+            const double C13_dd = 2.0 * (Constants::C13C12_MASSDIFF_U - 1.0);
+            Int iso1_index = getHighestIntensityPeakInMZRange(max_mono_peak.getMZ() + 1.0, spectrum1, 0, C13_dd);
+
+            if (iso1_index != -1)
+            {
+              const Peak1D& iso1_peak = spectrum1[iso1_index];
+              isotopes.push_back(iso1_peak);
+              Int iso2_index = getHighestIntensityPeakInMZRange(iso1_peak.getMZ() + 1.0, spectrum1, 0, C13_dd);
+
+              if (iso2_index != -1)
+              {
+                const Peak1D& iso2_peak = spectrum1[iso2_index];
+                isotopes.push_back(iso2_peak);
+              }
+            }
+          }
+        }
+
+        String query_id = String("unknown") + String(scan_index);
+
+        if (count == 0) //one ms file
+        {
+          // store data
+          String unique_name =  String(File::getUniqueName()).toQString(); //if not done this way - always new "unique name"
```

Comment:
maybe change comment. Do you want to say something like "generate unique name once"